### PR TITLE
Chat 266/filter illegal global channels

### PR DESCRIPTION
**Pull request by crgbaumgart**
- Added integration tests to validate globalChannel validator by an updated onRequest Function.
- See update Functions logic at https://github.com/pubnub/chat-engine-setup/pull/10
- FYI running tests requires updates to the onRequest Function

### CODE REVIEW

File: src/bootstrap.js [hunk 1]
```diff
@@ -156,9 +156,10 @@ module.exports = (ceConfig = {}, pnConfig = {}) => {
      * Parse a channel name into chat object parts
      * @private
      */
+     // TODO: DIS BISH GOTTA GO
```
Comment (ianjennings) [inline src/bootstrap.js]: wat?

File: src/bootstrap.js [hunk 2]
```diff
@@ -185,7 +186,8 @@ module.exports = (ceConfig = {}, pnConfig = {}) => {
         }
 
         if (channel.indexOf(ChatEngine.ceConfig.globalChannel) === -1) {
-            channel = [ChatEngine.ceConfig.globalChannel, 'chat', chanPrivString, channel].join('#');
+            // TODO: DIS BISH GOTTA GO
```
Comment (ianjennings) [inline src/bootstrap.js]: Add GH issues or Jira tickets instead of inline TODOs

File: src/bootstrap.js [hunk 3]
```diff
@@ -331,10 +333,11 @@ module.exports = (ceConfig = {}, pnConfig = {}) => {
      */
     ChatEngine.subscribeToPubNub = () => {
 
+        // TODO: DIS BISH GOTTA GO
         let chanGroups = [
-            ceConfig.globalChannel + '#' + ChatEngine.me.uuid + '#rooms',
-            ceConfig.globalChannel + '#' + ChatEngine.me.uuid + '#system',
-            ceConfig.globalChannel + '#' + ChatEngine.me.uuid + '#custom'
+            ceConfig.globalChannel + '_' + ChatEngine.me.uuid + '_rooms',
```
Comment (ianjennings) [inline src/bootstrap.js]: Did you rewrite the server component too?

File: src/bootstrap.js (hunk 2 above)
Comment (ianjennings) [inline src/bootstrap.js]: Add GH issues or Jira tickets instead of inline TODOs

File: test/integration/main.test.js [hunk 4]
```diff
@@ -272,6 +273,221 @@ describe('connect', () => {
...
+        ChatEngine.on('$.error.auth', (err) => {
+            assert.equal(err.error.response.status, 401);
+            const expected = 'Illegal ChatEngine `globalChannel`: ' + globalChannel + ' initialized.';
+            assert.equal(err.error.response.data, expected);
+            done();
+        });
+
+    });
+});
+
```
Comment (ianjennings) [inline test/integration/main.test.js]: so good :)

File: test/integration/main.test.js [hunk 5]
```diff
@@ -641,8 +866,7 @@ describe('invite', () => {
 
                 // me is the current context
                 yourChat.invite(ChatEngine.me);
-
-            }, 5000);
+            }, 2000);
```
Comment (ianjennings) [inline test/integration/main.test.js]: Not sure these are required to pass?

Comment (ianjennings) [reply test/integration/main.test.js]: I remember you mentioned it helped, but was probably unrelated. I did a lot of timeout config too when I dealt with similar problems.

File: src/bootstrap.js (hunk 3 above)
Comment (crgbaumgart) [reply src/bootstrap.js]: yes! See the recent merged changes to src/functions/server.js in chat-engine-setup repo.

File: src/bootstrap.js (hunk 2 above)
Comment (crgbaumgart) [reply src/bootstrap.js]: Removed

**Comment by crgbaumgart** [comment]
Hokay! @ianjennings FWIW chat-engine-setup repo has the changes pushed to master. We'll have to deploy another K8s pod with this updated server code. @ajb413 should have the knowledge on how this accomplished.

**Comment by ajb413** [comment]
@crgbaumgart https://github.com/pubnub/chat-engine-setup#deploy

**Comment by ianjennings** [comment]
@crgbaumgart This would also require that current customers migrate their functions. @dogaan @madisonsmith should be aware of this.

**Comment by crgbaumgart** [comment]
Its dead Jim.

**Pull request closed without merging**